mlflow/server/js/src/components/ParallelCoordinatesPlotView.js

```diff
@@ -147,9 +146,18 @@ export const generateAttributesForCategoricalDimension = (labels) => {
   return attributes;
 };
 
-// Infer type with the first run's value
+/**
+ * Infer the type of data in a run. If all the values are numbers or castable to numbers, then
+ * treat it as a number column.
+ */
 export const inferType = (key, runUuids, entryByRunUuid) => {
-  return isNaN(entryByRunUuid[runUuids[0]][key].value) ? 'string' : 'number';
+  for (let i = 0; i < runUuids.length; i++) {
+    const value = entryByRunUuid[runUuids[i]][key].value;
+    if (typeof(value) === "string" && isNaN(Number(value)) && value !== "NaN") {
```

Comment:
So we are getting 'NaN' string for NaN value?

Reply:
Yeah, this is in case people logged a numeric parameter that was NaN. Parameters are always stored as strings, so it would be stored as the string NaN, but if all the other values for the parameters are numbers, then we should still display that dimension as a number.